src/Microsoft.Azure.WebJobs.Logging/FunctionId.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs.Logging
+{
+    /// <summary>
+    /// Strongly typing for a function Id. 
+    /// FunctionIds include HostName and FunctionName
+    /// </summary>
+    public struct FunctionId
+    {
+        internal string Value { get; set; }
+
+        /// <summary>
+        /// Get string representation. This will be consist of valid Azure Table row key characters. 
+        /// </summary>
+        /// <returns></returns>
+        public override string ToString()
+        {
+            return this.Value;
+        }
+
+        /// <summary>
+        /// Throw exception if this instance is bad. 
+        /// </summary>
+        public void Validate()
+        {
+            if (string.IsNullOrEmpty(this.Value))
+            {
+                throw new InvalidOperationException("FunctionId is not valid.");
+            }
+        }
+
+        /// <summary>
+        /// Build directly from a string. This skips escaping and should only be used by the reader for interop. 
+        /// This is the direct inverse of ToString().
+        /// </summary>
+        /// <param name="value">raw value from a previos ToString() call. </param>
+        /// <returns></returns>
+        public static FunctionId DangerousFromString(string value)
+        {
+            return new FunctionId { Value = value };
+        }
+
+        /// <summary>
+        /// Build a FunctionId from the host and function name. 
+        /// </summary>
+        /// <param name="hostName"></param>
+        /// <param name="functionName"></param>
+        /// <returns></returns>
+        public static FunctionId Build(string hostName, string functionName)
```

Comment:
It's technically possible; but I avoided it because it loses semantics. Specifically, whether it escapes (Build) or doesn't escape (Parse).

---
In reply to: [84856006](https://github.com/Azure/azure-webjobs-sdk/pull/884#discussion_r84856006) [](ancestors = 84856006)